src/sys/socket/mod.rs

```diff
@@ -279,12 +277,18 @@ pub fn sendmsg<'a>(fd: RawFd, iov: &[IoVec<&'a [u8]>], cmsgs: &[ControlMessage<'
         None => (0 as *const _, 0),
     };
 
+    let cmsg_ptr = if capacity > 0 {
```

Comment:
I think, if anything, cmsg_buffer should be allocated with `capacity` rather than `len`.